core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2755,6 +2755,31 @@ private static Integer getJsonDepth(Object o) {
     return depth;
   }
 
+  public static Integer jsonLength(Object input) {
+    final PathContext context = (PathContext) input;
+    final Integer result;
+    try {
+      Object value = context.pathReturned;
```

Comment:
`context` may carries an error (typically when user passes a "strict" jsonpath expression). Let's add a fail-fast condition to exclude that. 
E.g.
```java
if (context.exc != null) {
  throw toUnchecked(context.exc);
}
try {
  ...
```